### ISSUE TITLE
getID3 replaces "gs://" with "gs:/" in Google Cloud Storage filenames leading file not found errors

### ISSUE THREAD

**Issue by iqbalansari** [JamesHeinrich/getID3#43]
Hi,

I was trying to integrate getID3 in a google app engine project, I am currently working on. We use [google cloud storage](https://cloud.google.com/appengine/docs/php/googlestorage/) for storing audio files. The PHP runtime on appengine has the Google Cloud Storage stream wrapper built in, thus allowing use of standard PHP file system functions to access the objects using the 'gs://' stream.

However getID3 [replaces](https://github.com/JamesHeinrich/getID3/blob/ed9e77959015d61bf93fe144d552f27d58397099/getid3/getid3.php#L267) the two forward slashes in 'gs://' with a single slash thus making the file path invalid and eventually failing (since it could not find the file). 

I am using (if it is relevant by any chance)

- getID3 v1.9.9
- PHP 5.5 running on Google app engine

Thanks

**Comment by JamesHeinrich** [comment]
Not a getID3 issue.

getID3 does not support remote files, whether over http:// or gs:// or your other favourite protocol. You'll have to manage your own wrapper functions to interact with the remote filesystem and let getID3 deal only with local files.

**Comment by soulhunter1987** [comment]
Yes, it is remote file, but due to PHP-builtin support it can be accessed with standard functions like fopen, fseek, fread, etc. So it is supported. We just need to fix a single function which validates path. It will be done in v 2.0

**Comment by soulhunter1987** [comment]
@iqbalansari ,  If you want to use v 1.9 - you'll have to modify souce code yourself.

**Comment by maknapp** [comment]
I was just trying to get this to work with a file wrapper and there are more things to change than just the path validation. Several functions (like `is_file()`) are used in getID3 that do not work with file wrappers.

**Comment by iqbalansari** [comment]
@maknapp As far as Google Cloud Storage is concerned `is_file` [is supported](https://cloud.google.com/appengine/docs/php/googlestorage/advanced#php_filesystem_functions_support_on_google_cloud_storage). I needed to set `option_max_2gb_check` to false, to prevent getID3 for trying to get file size using shell commands like `ls` (these are not supported on GAE), in this case I am absolutely sure the files I am processing would be far smaller than 2GB.

Except for the issue mentioned in OP I have been able to use getID3 on GAE without any issues. I hack around the this issue right now by passing `gs:///` to getID3 instead of just `gs://`, of course this is just a hack, it would be great if this can be fixed in getID3 itself

@soulhunter1987  I do not mind switching to v2.0, I preferred v1.9 because there is no stable release for v2.0 on packagist. Thanks for looking into this.
